src/main/java/com/github/maxopoly/finale/listeners/PearlCoolDownListener.java

```diff
@@ -39,45 +42,57 @@ public static long getPearlCoolDown(UUID uuid) {
 	private boolean combatTag;
 	private boolean setVanillaCooldown;
 	private boolean useSideBar;
-	private CivScoreBoard scoreBoard;
-	private Set<UUID> onCooldown;
+	private boolean useActionBar;
 
 	public PearlCoolDownListener(long cooldown, boolean combatTag, CombatTagPlusManager ctpManager,
-			boolean setVanillaCooldown, boolean useSideBar) {
+			boolean setVanillaCooldown, boolean useSideBar, boolean useActionBar) {
 		instance = this;
 		this.cds = new TickCoolDownHandler<UUID>(Finale.getPlugin(), cooldown);
 		this.ctpManager = ctpManager;
 		this.combatTag = combatTag;
 		this.setVanillaCooldown = setVanillaCooldown;
 		this.useSideBar = useSideBar;
+		this.useActionBar = useActionBar;
+	}
+
+	public long getCoolDown() {
+		return cds.getTotalCoolDown();
+	}
+	
+	@EventHandler
+	public void onJoin(PlayerJoinEvent e) {
+		Player player = e.getPlayer();
+		
+		UI ui = UIManager.getUIManager().getScoreboard(player);
 		if (useSideBar) {
-			onCooldown = Collections.synchronizedSet(new TreeSet<>());
-			scoreBoard = ScoreBoardAPI.createBoard("finalePearlCoolDown");
-			Bukkit.getScheduler().scheduleSyncRepeatingTask(Finale.getPlugin(), () -> {
-				Iterator<UUID> iter = onCooldown.iterator();
-				while (iter.hasNext()) {
-					UUID uuid = iter.next();
-					Player p = Bukkit.getPlayer(uuid);
-					if (!cds.onCoolDown(uuid)) {
-						iter.remove();
-						if (p != null) {
-							scoreBoard.hide(p);
-						}
-						continue;
+			ui.getUIHandlers().add(new UIHandler() {
```

Comment:
That API seems straight up worse to me, curious to hear advantage of it

Reply:
With your API, don't seem to be able to choose where the item should go on the scoreboard.

Reply:
Yes, that's intended and necessary. If you have multiple plugins adding entries, they won't know what already exists and not. Finale shouldn't have to check whether Citadel, NameLayer, Bastion etc. are enabled and what entries they are possibly using. Instead a central API dynamically handles that assignment and enables independent usage of the API. Requiring individual plugins to manage slots is not acceptable.

If anything we can change my API to center entries or allow a "preferred" slot, but tbh I don't see the point in even that. Wouldn't a box as small as possible be preferable?

Reply:
I think having a consistent order of scoreboard items is beneficial, especially when items pertain to PvP. As more plugins and features utilize the scoreboard, the scoreboard grows in size and the order becomes very important. Consider a change to your HUD like moving potion effects from one side of the screen to other or relocating your radar; it takes a while to get used to it, and after you've got used to it, you don't think about where to look anymore.

Reply:
Well, the API should allow specifying a priority or something like that then, but again leaving order assignment to using plugins is entirely unacceptable.

I'll make some changes to my API to add something that should kind of address this.